AsyncDisplayKit/TextKit/ASTextKitFontSizeAdjuster.mm

```diff
@@ -179,23 +181,42 @@ - (CGFloat)scaleFactor
     
     if (_attributes.maximumNumberOfLines > 0) {
       // get the number of lines in our possibly scaled string
-      NSUInteger numberOfLines = [self lineCountForString:textStorage];
+      NSUInteger numberOfLines = [self lineCountForString:scaledString];
       if (numberOfLines > _attributes.maximumNumberOfLines) {
         
         for (NSUInteger index = scaleIndex; index < scaleFactors.count; index++) {
-          NSMutableAttributedString *entireAttributedString = [[NSMutableAttributedString alloc] initWithAttributedString:textStorage];
-          [[self class] adjustFontSizeForAttributeString:entireAttributedString withScaleFactor:[scaleFactors[index] floatValue]];
-          
+          scaledString = [[NSMutableAttributedString alloc] initWithAttributedString:textStorage];
           
           // save away this scale factor. Even if we don't fit completely we should still scale down
           adjustedScale = [scaleFactors[index] floatValue];
+          [[self class] adjustFontSizeForAttributeString:scaledString withScaleFactor:adjustedScale];
+          
+          // adjust here so we start at the proper place in our scale array if we are too tall
+          scaleIndex++;
           
-          if ([self lineCountForString:entireAttributedString] <= _attributes.maximumNumberOfLines) {
+          if ([self lineCountForString:scaledString] <= _attributes.maximumNumberOfLines) {
             // we fit! we are done
             break;
           }
         }
+      }
+    }
+    
+    // check to see if the string's height is greater that the constrained height
+    CGSize stringSize = [scaledString boundingRectWithSize:CGSizeMake(_constrainedSize.width, CGFLOAT_MAX) options:NSStringDrawingUsesLineFragmentOrigin context:nil].size;
```

Comment:
We currently use NSLayoutManager and NSTextContainer via ASTextKitRenderer to size strings. Any way we can use ASTextKitRenderer to do this?